x-pack/test/functional_basic/apps/ml/permissions/full_ml_access.ts

```diff
@@ -0,0 +1,163 @@
+          await ml.testExecution.logTestStep('should load the data visualizer page');
+          await ml.navigation.navigateToDataVisualizer();
+
+          await ml.testExecution.logTestStep(
+            'should display the "import data" card with enabled button'
+          );
+          await ml.dataVisualizer.assertDataVisualizerImportDataCardExists();
+          await ml.dataVisualizer.assertUploadFileButtonEnabled(true);
+
+          await ml.testExecution.logTestStep(
+            'should display the "select index pattern" card with enabled button'
+          );
+          await ml.dataVisualizer.assertDataVisualizerIndexDataCardExists();
+          await ml.dataVisualizer.assertSelectIndexButtonEnabled(true);
+
+          await ml.testExecution.logTestStep(
+            'should display the "start trial" card with enabled button'
+          );
+          await ml.dataVisualizer.assertDataVisualizerStartTrialCardExists();
+          await ml.dataVisualizer.assertStartTrialButtonEnabled(true);
+        });
+
+        it('should display elements on Index Data Visualizer page correctly', async () => {
+          await ml.testExecution.logTestStep('should load an index into the data visualizer page');
+          await ml.dataVisualizer.navigateToIndexPatternSelection();
+          await ml.jobSourceSelection.selectSourceForIndexBasedDataVisualizer(ecIndexPattern);
+
+          await ml.testExecution.logTestStep('should display the time range step');
+          await ml.dataVisualizerIndexBased.assertTimeRangeSelectorSectionExists();
+
+          await ml.testExecution.logTestStep('should load data for full time range');
+          await ml.dataVisualizerIndexBased.clickUseFullDataButton(ecExpectedTotalCount);
+
+          await ml.testExecution.logTestStep('should display the panels of fields');
+          await ml.dataVisualizerIndexBased.assertFieldsPanelsExist(ecExpectedFieldPanelCount);
+
+          await ml.testExecution.logTestStep('should not display the actions panel with cards');
+          await ml.dataVisualizerIndexBased.assertActionsPanelNotExists();
+          await ml.dataVisualizerIndexBased.assertCreateAdvancedJobCardNotExists();
+          await ml.dataVisualizerIndexBased.assertRecognizerCardNotExists(ecExpectedModuleId);
+        });
+
+        it('should display elements on File Data Visualizer page correctly', async () => {
+          await ml.testExecution.logTestStep('should load the file data visualizer file selection');
+          await ml.navigation.navigateToDataVisualizer();
+          await ml.dataVisualizer.navigateToFileUpload();
+
+          await ml.testExecution.logTestStep(
+            'should select a file and load visualizer result page'
+          );
+          await ml.dataVisualizerFileBased.selectFile(uploadFilePath);
+
+          await ml.testExecution.logTestStep('should displays components of the file details page');
```

Comment:
Typo - `should display` (not `displays`)

Reply:
Fixed in c4b15a6